classes/observers/events/PKPUsageEvent.inc.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+/**
+ * @file classes/observers/events/PKPUsageEvent.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PKPUsageEvent
+ * @ingroup observers_events
+ *
+ * @brief Usage event.
+ *
+ */
+
+namespace PKP\observers\events;
+
+use APP\core\Application;
+use APP\statistics\StatisticsHelper;
+use Illuminate\Foundation\Events\Dispatchable;
+use PKP\config\Config;
+use PKP\core\Core;
+use PKP\db\DAORegistry;
+
+class PKPUsageEvent
+{
+    use Dispatchable;
+
+    /** Current time */
+    public string $time;
+    /** User IP adress */
+    public string $ip;
+    /** Canonical URL for the pub object */
+    public string $canonicalUrl;
+    /** Context ID */
+    public int $contextId;
+    /** Submission ID */
+    public ?int $submissionId;
+    /** Representation (galley or publication format) ID */
+    public ?int $representationId;
+    /** Viewed/downloaded object, one of the Application::ASSOC_TYPE_... constants */
+    public int $assocType;
+    /** Viewed/downloaded object ID */
+    public int $assocId;
+    /** COUNTER file type, one of the StatisticsHelper::STATISTICS_FILE_TYPE_... constants*/
+    public ?int $fileType;
+    /** User's UserAgent */
+    public string $userAgent;
+    /** Application's complete version string */
+    public string $version;
+
+    /**
+     * Create a new usage event instance.
+     */
+    public function __construct(int $assocType, int $assocId, int $contextId, int $submissionId = null, int $representationId = null, string $mimetype = null)
+    {
+        $application = Application::get();
+        $request = $application->getRequest();
```

Comment:
Because this is the usage event constructor, that is called when viewing/using an object, and has for now nothing to do with listener, the time is correctly defined here, I think -- i.e. in the event and not in the listener -- but yes, I would then include the request object as event attribute...